In [ ]:
#1. Imprimir los números del 1 al 5 utilizando un bucle for
Descripción: Utiliza un bucle for para imprimir los números del 1 al 5 (incluido) en orden ascendente.

In [1]:
for x in range(6):
    print(x)

0
1
2
3
4
5


2. Calcular la suma de los primeros 10 números naturales utilizando un bucle for
Descripción: Utiliza un bucle for para calcular la suma de los primeros 10 números naturales (1 + 2 + 3 + ... + 10).

In [15]:
lista_naturales = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
suma = 0

for num in lista_naturales:
    suma = suma + num
print(suma)

55


In [22]:
# Opción de Carla

suma = 0  
for num in range(1, 11):  
    suma = suma + num
print(suma)

55


In [ ]:
# 3. Calcular el producto de los elementos de una lista
# Descripción: Dada una lista de números, utiliza un bucle for para calcular el producto de todos los elementos.

In [23]:
lista_2 = [2, 5, 8]
producto = 1
for num in lista_2:
    producto = producto * num
print(producto)


80


In [25]:
# Opción de Carla con un range de 1-11
producto = 1  
for num in range(1, 11):  
    producto = producto * num 
print(producto)

3628800


In [ ]:
# 4. Imprimir los caracteres de una cadena de texto
# Descripción: Dada una cadena de texto, utiliza un bucle for para imprimir cada carácter en una línea separada.

In [27]:
nuestra_frase = 'Python es una pesadilla'
for l in nuestra_frase:
    print(l)

P
y
t
h
o
n
 
e
s
 
u
n
a
 
p
e
s
a
d
i
l
l
a


In [ ]:
# 5. Imprimir los números pares del 1 al 100 utilizando un bucle for
# Descripción: Utiliza un bucle for para imprimir los números pares del 1 al 100 en orden ascendente.

In [31]:
for x in range(0,101,2):
    print(x)

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100


In [ ]:
# 6. Imprimir los números primos del 1 al 50 utilizando un bucle for
# Descripción: Utiliza un bucle for para imprimir los números primos del 1 al 50 en orden ascendente.
# NOTA: Según la Wikipedia, un número primo es un número natural mayor que 1 que no tiene divisores positivos más que 1 y él mismo.

In [1]:
for numero in range(1, 51): #1
    if numero > 1:  #2
        es_primo = True  # 3
        for i in range(2, numero):  # 4
            if numero % i == 0:  # 5
                es_primo = False  # 6
                break  # 7
        if es_primo:  # 8
            print(numero)  # 9

2
3
5
7
11
13
17
19
23
29
31
37
41
43
47


for numero in range(1, 51): #1
    if numero > 1:  #2
        es_primo = True  # 3
        for i in range(2, numero):  # 4
            if numero % i == 0:  # 5
                es_primo = False  # 6
                break  # 7
        if es_primo:  # 8
            print(numero)  # 9
Empezamos usando el bucle for, con el objetivo de iterar por nuestro rango de números (en este caso hemos escogido del 1 al 51)
Para empezar a desarrollar lo que nos pide el ejercicio vamos a empezar a establecer condiciones con un if. La primera de ellas será que solo consideramos números mayores que 1, ya que 1 no se considera un número primo.
Al pasar nuestra primera condición, vamos a suponer que el número es primero hasta que se demuestre lo contrario, para ello declararemos la variable "es_primo" y la igualaremos a la palabra reservada "True"
Vamos a hacer otro bucle, lo haremos dentro del anterior para seguir averiguando la información que queremos extraer. Así que vamos a hacer otro for para recorrer esa lista de números, uno por uno, asignando temporalmente a "i" el valor de cada número en la lista. Usaremos esto para verificar si "numero" se puede dividir además de por 1 y por si mismo, por más números... Para lograrlo esta vez vamos a iterar desde 2 hasta "número" (recordamos que range incluye el primer número pero excluye el último  por lo que vamos a entenderlo como "numero -1". Por ej: Por ejemplo, si número es 5, range(2, numero) crea una lista de números que comienza en 2 y termina en 4 (es decir, [2, 3, 4]
Una vez tenemos esa iteración, vamos a pasar a ponerle la siguiente de nuestras condiciones, en este caso será la verificación de su divisibilidad -> Si "numero" es divisible por "i" (numero % i == 0) entonces "numero" no es primo.
Esta vez dado el paso anterior, si encontramos un divisor entonces vamos a cambiar el estado de nuestra variable "es primo" a la palabra reservada "False"
Ya casi estamos! Ahora vamos a usar el comando break para salir de nuestro bucle interior (segundo bucle) ya que no  vamos a necesitar buscar más divisores
Hacemos un verificación final... Después de verificar todos los posibles divisores, si "es_primo" aún es "True", entonces el número es primo.
Y ahora si, después de verificar todos los posibles divisores, el código va a ejecutarse dándonos como resultado una lista con los números primos del 1 al 50 

In [ ]:
# 7. Calcular el factorial de un número utilizando un bucle for
# Descripción: Dado un número entero positivo, calcula su factorial utilizando un bucle for.

In [3]:
numero = 5 #1
factorial = 1 #2
for i in range(1, numero+1): #3
    factorial *= i #4
print("El factorial de", numero, "es:", factorial) #5

El factorial de 5 es: 120


numero = 5 #1
factorial = 1 #2
for i in range(1, numero+1): #3
    factorial *= i #4
print("El factorial de", numero, "es:", factorial) #5
Iniciamos la resolución de nuestro ejercicio declarando una variable con el número que queremos escoger para calcular su factorial
En este paso vamos a declarar otra variable llamada factorial y le daremos como valor 1. Esto es importante porque el factorial de un número se calcula multiplicando una serie de números, y necesitamos empezar con el valor 1 (ya que 1 es el elemento neutro de la multiplicación).
Como siempre que queramos iterar vamos a hacer un bucle for para recorrer un rango de que abarca desde 1 hasta "numero" (Recordamos! Range excluye el último número, como nosotras vamos a querer incluirlo, procedemos a poner numero+1, En este caso, como "numero" es 5, el bucle va a iterar sobre los números 1, 2, 3, 4, y 5.)
Dentro del bucle, vamos multiplicando el valor actual de factorial por "i", que es el número actual en nuestra iteración. Esto significa que en cada paso del bucle, multiplicamos el valor acumulado de "factorial" por el siguiente número en nuestra secuencia hasta llegar a 5. Por ejemplo, empezamos con 1 (valor inicial de factorial), luego 12, luego 123, luego 1234, y finalmente 12345.
Lo logramos! Una vez que hemos multiplicado todos los números hasta 5, ya sólo nos queda imprimir el resultado que ente caso no dará: 120